confluent_kafka/schema_registry/schema_registry_client.py

```diff
@@ -220,20 +340,12 @@ def get_registration(self, subject_name, schema):
             https://docs.confluent.io/current/schema-registry/develop/api.html#post--subjects-(string-%20subject)
 
         """  # noqa: E501
-        version = schema.subjects.get(subject_name, None)
-        if version:
-            return version
-
-        return self._rest_client.POST(
-            url_formatter(SUBJECT, subject_name),
-            encoder=_SchemaJSONEncoder,
-            decoder=_VersionJSONDecoder,
-            schema=schema)
+        response = self._rest_client.POST('subjects/' + _urlencode(subject_name),
+                                          body=schema)
+        return RegisteredSchema.from_json(response)
 
     def get_subjects(self):
```

Comment:
This supposedly returns the same thing as get_registration() (but all of them), so maybe the method names should match?